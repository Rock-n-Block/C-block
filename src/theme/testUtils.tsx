import React from 'react';

import { ThemeProvider as MuiThemeProvider } from '@material-ui/core';

import { theme } from './theme';

export const ThemeProvider: React.FC = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
);
