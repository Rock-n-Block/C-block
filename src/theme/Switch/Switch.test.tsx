import React from "react";

import { Switch } from "@material-ui/core";
import { render } from "@testing-library/react";

import { ThemeProvider } from "../testUtils";

describe("Switch", () => {
  it("should render", () => {
    const { container } = render(
      <ThemeProvider>
        <Switch checked />
        <Switch />
        <Switch checked disabled />
        <Switch disabled />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
