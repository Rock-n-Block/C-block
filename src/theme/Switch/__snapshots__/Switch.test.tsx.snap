// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render 1`] = `
<div>
  <span
    class="MuiSwitch-root"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-2 Mui-checked"
      data-testautomation-id="switch"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input-4 MuiSwitch-input"
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
  <span
    class="MuiSwitch-root"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorPrimary"
      data-testautomation-id="switch"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4 MuiSwitch-input"
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
  <span
    class="MuiSwitch-root"
  >
    <span
      aria-disabled="true"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-2 Mui-checked PrivateSwitchBase-disabled-3 Mui-disabled Mui-disabled Mui-disabled"
      data-testautomation-id="switch"
      tabindex="-1"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input-4 MuiSwitch-input"
          disabled=""
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
  <span
    class="MuiSwitch-root"
  >
    <span
      aria-disabled="true"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-disabled-3 Mui-disabled Mui-disabled Mui-disabled"
      data-testautomation-id="switch"
      tabindex="-1"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4 MuiSwitch-input"
          disabled=""
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
</div>
`;
